src/AsmResolver.DotNet/ModuleDefinition.cs

```diff
@@ -1295,14 +1295,18 @@ public void Write(IBinaryStreamWriter writer, IPEImageBuilder imageBuilder, IPEF
         public IPEImage ToPEImage(IPEImageBuilder imageBuilder)
         {
             var result = imageBuilder.CreateImage(this);
-            if (result.DiagnosticBag.HasErrors)
+            if (result.ErrorListener is DiagnosticBag diagnosticBag && diagnosticBag.HasErrors)
             {
                 throw new AggregateException(
                     "Construction of the PE image failed with one or more errors.",
-                    result.DiagnosticBag.Exceptions);
+                    diagnosticBag.Exceptions);
+            }
+            else if (result.HasFailed)
+            {
+                throw new Exception("Construction of the PE image failed.");
```

Comment:
Let's not use the generic `Exception`, but maybe use `MetadataBuilderException` instead, or else define a new exception type (maybe `PEImageBuilderException`?).